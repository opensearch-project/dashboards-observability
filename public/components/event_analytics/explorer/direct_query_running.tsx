/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { useDispatch } from 'react-redux';
import { EuiProgress, EuiEmptyPrompt, EuiButton } from '@elastic/eui';
import { update as updateSearchMetaData } from '../redux/slices/search_meta_data_slice';

export const DirectQueryRunning = ({ tabId }: { tabId: string }) => {
  const dispatch = useDispatch();
  return (
    <EuiEmptyPrompt
      icon={<EuiProgress size="xs" color="accent" />}
      title={<h2>Query Processing</h2>}
      body={
        <EuiButton
          color="success"
          onClick={() => {
            dispatch(
              updateSearchMetaData({
                tabId,
                data: {
                  isPolling: false,
                },
              })
            );
          }}
        >
          Cancel
        </EuiButton>
      }
    />
  );
};
