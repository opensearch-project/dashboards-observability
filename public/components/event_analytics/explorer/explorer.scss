/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

.liveStream {
  margin: 8px;
  height: 40px;
  align-items: center;
  justify-content: center;
  flex-direction: row;
  display: flex;
  flex-grow: 1;
  vertical-align: baseline;
}

.mainContentTabs .euiResizableContainer {
  height: calc(100vh - 194px);
}

.explorer-loading-spinner {
  position: relative;
  left: 50%;
  top: 50vh;
  width: 20px;
  height: 20px;
  margin-left: -5vw;
  margin-top: -20vh;
}
