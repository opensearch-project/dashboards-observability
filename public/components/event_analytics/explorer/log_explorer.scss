/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

.queryTabs {
  .euiTabs {
    .tab-title {
      max-width: 8rem;
      overflow: hidden;
      white-space: nowrap;
      text-overflow: ellipsis;
    }
    svg {
      vertical-align: inherit;
    }
    .linkNewTag {
      display: inline-block;
      text-align: center;
      font-size: 0.875rem;
      line-height: $tab-new-line-height;
      padding: $tab-new-padding; // align with content tab with small size 
      min-width: 6rem;
    }
  }
}

.tab-title {
  display: inline-block;
  margin-right: 5px;
}

.search-area {
  position: relative;
}