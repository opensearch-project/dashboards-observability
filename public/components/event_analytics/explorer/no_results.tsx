/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { FormattedMessage } from '@osd/i18n/react';
import {
  EuiCallOut,
  EuiFlexGroup,
  EuiFlexItem,
  EuiPage,
  EuiSpacer,
  EuiText,
  EuiEmptyPrompt,
} from '@elastic/eui';
import { useSelector } from 'react-redux';
import { coreRefs } from '../../../framework/core_refs';
import { selectQueries } from '../redux/slices/query_slice';

export const NoResults = ({ tabId }: any) => {
  // get the queries isLoaded, if it exists AND is true = show no res
  const queryInfo = useSelector(selectQueries)[tabId];

  return (
    <EuiPage paddingSize="s">
      {coreRefs.queryAssistEnabled ? (
        <>
          {/* check to see if the rawQuery is empty or not */}
          {queryInfo?.rawQuery ? (
            <EuiFlexGroup justifyContent="center" direction="column">
              <EuiFlexItem grow={false}>
                <EuiCallOut
                  title={
                    <FormattedMessage
                      id="observability.noResults.noResultsMatchSearchCriteriaTitle"
                      defaultMessage="No results match your search criteria"
                    />
                  }
                  color="warning"
                  iconType="help"
                  data-test-subj="observabilityNoResultsCallout"
                />
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiEmptyPrompt
                  iconType={'editorCodeBlock'}
                  title={<h2>No results</h2>}
                  body={
                    <p>
                      Try selecting a different data source, expanding your time range or modifying
                      the query & filters. You may also use the Query Assistant to fine-tune your
                      query using simple conversational prompts.
                    </p>
                  }
                />
              </EuiFlexItem>
            </EuiFlexGroup>
          ) : (
            <EuiEmptyPrompt
              iconType={'editorCodeBlock'}
              title={<h2>Get started</h2>}
              body={
                <p>
                  Run a query to view results, or use the Query Assistant to automatically generate
                  complex queries using simple conversational prompts.
                </p>
              }
            />
          )}
        </>
      ) : (
        <EuiFlexGroup justifyContent="center" direction="column">
          <EuiFlexItem grow={false}>
            <EuiCallOut
              title={
                <FormattedMessage
                  id="observability.noResults.noResultsMatchSearchCriteriaTitle"
                  defaultMessage="No results match your search criteria"
                />
              }
              color="warning"
              iconType="help"
              data-test-subj="observabilityNoResultsCallout"
            />
          </EuiFlexItem>
          <EuiFlexItem grow={false}>
            <EuiSpacer size="s" />
            <EuiText>
              <h2 data-test-subj="obsNoResultsTimefilter">
                <FormattedMessage
                  id="observability.noResults.expandYourTimeRangeTitle"
                  defaultMessage="Select a data source, expand your time range, or modify the query"
                />
              </h2>
              <p>
                <FormattedMessage
                  id="observability.noResults.queryMayNotMatchTitle"
                  defaultMessage="After selection, check the time range, query filters, fields, and query"
                />
              </p>
            </EuiText>
          </EuiFlexItem>
        </EuiFlexGroup>
      )}
    </EuiPage>
  );
};
