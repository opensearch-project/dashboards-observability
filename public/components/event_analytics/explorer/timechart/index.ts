/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export { HitsCounter } from './hits_counter/hits_counter';
export { Timechart } from './timechart';
export { getTimeRangeFromCountDistribution } from './utils';
