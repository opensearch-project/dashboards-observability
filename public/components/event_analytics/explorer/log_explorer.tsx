/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
/* eslint-disable react-hooks/exhaustive-deps */
import { isEmpty } from 'lodash';
import React, { useContext, useEffect, useRef, useState } from 'react';
import { useSelector } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { LogExplorerRouterContext } from '..';
import {
  APP_ANALYTICS_TAB_ID_REGEX,
  CREATE_TAB_PARAM_KEY,
  TAB_CHART_ID,
  TAB_EVENT_ID,
} from '../../../../common/constants/explorer';
import { EmptyTabParams, ILogExplorerProps } from '../../../../common/types/explorer';
import { selectQueryResult } from '../redux/slices/query_result_slice';
import { selectQueries } from '../redux/slices/query_slice';
import { selectQueryTabs } from '../redux/slices/query_tab_slice';
import { Explorer } from './explorer';
import { getDateRange } from '../utils/utils';
import {
  QUERY_ASSIST_END_TIME,
  QUERY_ASSIST_START_TIME,
} from '../../../../common/constants/shared';
import { coreRefs } from '../../../../public/framework/core_refs';

const searchBarConfigs = {
  [TAB_EVENT_ID]: {
    showSaveButton: true,
    showSavePanelOptionsList: false,
  },
  [TAB_CHART_ID]: {
    showSaveButton: true,
    showSavePanelOptionsList: true,
  },
};

const getExistingEmptyTab = ({ tabIds }: EmptyTabParams) => tabIds[0];

export const LogExplorer = ({
  pplService,
  dslService,
  savedObjects,
  timestampUtils,
  setToast,
  savedObjectId,
  notifications,
  http,
  queryManager,
  dataSourcePluggables,
}: ILogExplorerProps) => {
  const history = useHistory();
  const routerContext = useContext(LogExplorerRouterContext);
  const tabIds = useSelector(selectQueryTabs).queryTabIds.filter(
    (tabid: string) => !tabid.match(APP_ANALYTICS_TAB_ID_REGEX)
  );
  const queries = useSelector(selectQueries);
  const curSelectedTabId = useSelector(selectQueryTabs).selectedQueryTab;
  const explorerData = useSelector(selectQueryResult);
  const queryRef = useRef();
  const tabIdsRef = useRef();
  const explorerDataRef = useRef();
  const curSelectedTabIdRef = useRef();
  queryRef.current = queries;
  tabIdsRef.current = tabIds;
  explorerDataRef.current = explorerData;
  curSelectedTabIdRef.current = curSelectedTabId;

  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const [tabCreatedTypes, setTabCreatedTypes] = useState({});

  const dispatchSavedObjectId = async () => {
    return getExistingEmptyTab({
      tabIds: tabIdsRef.current,
      queries: queryRef.current,
      explorerData: explorerDataRef.current,
    });
  };

  const dateRange = getDateRange(undefined, undefined, queries[tabIds[0]]);
  const [startTime, setStartTime] = useState(dateRange[0]);
  const [endTime, setEndTime] = useState(dateRange[1]);

  useEffect(() => {
    if (!isEmpty(savedObjectId)) {
      dispatchSavedObjectId();
    }

    if (routerContext && routerContext.searchParams.has(CREATE_TAB_PARAM_KEY)) {
      // need to wait for current redux event loop to finish
      setImmediate(() => {
        routerContext.searchParams.delete(CREATE_TAB_PARAM_KEY);
        routerContext.routerProps.history.replace({
          search: routerContext.searchParams.toString(),
        });
      });
    }
  }, []);

  return (
    <>
      <Explorer
        key={`explorer_${tabIds[0]}`}
        pplService={pplService}
        dslService={dslService}
        tabId={tabIds[0]}
        savedObjects={savedObjects}
        timestampUtils={timestampUtils}
        setToast={setToast}
        history={history}
        notifications={notifications}
        savedObjectId={savedObjectId}
        tabCreatedTypes={tabCreatedTypes}
        curSelectedTabId={curSelectedTabIdRef}
        http={http}
        searchBarConfigs={searchBarConfigs}
        startTime={startTime}
        endTime={endTime}
        setStartTime={setStartTime}
        setEndTime={setEndTime}
        queryManager={queryManager}
        dataSourcePluggables={dataSourcePluggables}
      />
    </>
  );
};
