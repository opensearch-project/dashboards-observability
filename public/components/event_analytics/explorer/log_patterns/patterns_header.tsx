/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import {
  EuiButton,
  EuiButtonEmpty,
  EuiSmallButtonIcon,
  EuiCompressedFieldText,
  EuiFlexGroup,
  EuiFlexItem,
  EuiCompressedFormRow,
  EuiLink,
  EuiPopover,
  EuiPopoverFooter,
  EuiSpacer,
  EuiText,
  EuiTitle,
} from '@elastic/eui';
import { PatternTableData } from 'common/types/explorer';
import { PPL_PATTERNS_DOCUMENTATION_URL } from '../../../../../common/constants/shared';

interface PatternsHeaderProps {
  patternsData: { patternTableData: PatternTableData[] };
  patternRegexInput: string;
  setPatternRegexInput: (pattern: string) => void;
  onPatternApply: () => void;
  setIsPatternConfigPopoverOpen: (flag: boolean) => void;
  isPatternConfigPopoverOpen: boolean;
}

export const PatternsHeader = ({
  patternsData,
  patternRegexInput,
  setPatternRegexInput,
  onPatternApply,
  setIsPatternConfigPopoverOpen,
  isPatternConfigPopoverOpen,
}: PatternsHeaderProps) => {
  return (
    <EuiFlexGroup gutterSize="s" alignItems="center">
      <EuiFlexItem grow={false}>
        <EuiTitle size="s">
          <h3 style={{ margin: '0px' }}>
            Patterns{' '}
            <span className="pattern-header-count">
              ({patternsData.patternTableData?.length || 0})
            </span>
          </h3>
        </EuiTitle>
      </EuiFlexItem>
      <EuiFlexItem grow={false}>
        <EuiPopover
          button={
            <EuiSmallButtonIcon
              iconType="gear"
              onClick={() => setIsPatternConfigPopoverOpen(!isPatternConfigPopoverOpen)}
            />
          }
          isOpen={isPatternConfigPopoverOpen}
          closePopover={() => setIsPatternConfigPopoverOpen(false)}
          anchorPosition="upCenter"
        >
          <EuiTitle size="xxs">
            <h3>Pattern regex</h3>
          </EuiTitle>
          <EuiText size="s">Log patterns allow you to cluster your logs, to help</EuiText>
          <EuiText size="s">summarize large volume of logs.</EuiText>
          <EuiSpacer size="s" />
          <EuiCompressedFormRow
            helpText={
              <EuiText size="s">
                Pattern regex is used to reduce logs into log groups.{' '}
                <EuiLink href={PPL_PATTERNS_DOCUMENTATION_URL} target="_blank">
                  help
                </EuiLink>
              </EuiText>
            }
          >
            <EuiCompressedFieldText
              value={patternRegexInput}
              onChange={(e) => setPatternRegexInput(e.target.value)}
            />
          </EuiCompressedFormRow>
          <EuiPopoverFooter>
            <EuiFlexGroup justifyContent="flexEnd">
              <EuiFlexItem grow={false}>
                <EuiButtonEmpty size="s" onClick={() => setIsPatternConfigPopoverOpen(false)}>
                  Cancel
                </EuiButtonEmpty>
              </EuiFlexItem>
              <EuiFlexItem grow={false}>
                <EuiButton
                  size="s"
                  fill
                  onClick={() => {
                    setIsPatternConfigPopoverOpen(false);
                    onPatternApply();
                  }}
                >
                  Apply
                </EuiButton>
              </EuiFlexItem>
            </EuiFlexGroup>
          </EuiPopoverFooter>
        </EuiPopover>
      </EuiFlexItem>
    </EuiFlexGroup>
  );
};
