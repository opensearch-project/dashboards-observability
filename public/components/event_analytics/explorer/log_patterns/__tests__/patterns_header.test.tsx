/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { waitFor } from '@testing-library/react';
import { PatternsHeader } from '../patterns_header';
import { sampleLogPatternData } from '../../../../../../common/constants/explorer';

describe('Patterns header component', () => {
  configure({ adapter: new Adapter() });

  it('Renders header of log patterns', async () => {
    const patternRegexInput = '[a-zA-Zd]';
    const patternsData = { patternTableData: [{ ...sampleLogPatternData }] };

    const wrapper = mount(
      <PatternsHeader
        patternsData={patternsData}
        patternRegexInput={patternRegexInput}
        setPatternRegexInput={jest.fn()}
        onPatternApply={jest.fn()}
        setIsPatternConfigPopoverOpen={jest.fn()}
        isPatternConfigPopoverOpen={true}
      />
    );

    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
