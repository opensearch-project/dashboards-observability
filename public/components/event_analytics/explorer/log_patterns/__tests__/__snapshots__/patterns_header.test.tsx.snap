// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Patterns header component Renders header of log patterns 1`] = `
<div
  className="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
>
  <div
    className="euiFlexItem euiFlexItem--flexGrowZero"
  >
    <h3
      className="euiTitle euiTitle--small"
      style={
        Object {
          "margin": "0px",
        }
      }
    >
      Patterns
       
      <span
        className="pattern-header-count"
      >
        (
        1
        )
      </span>
    </h3>
  </div>
  <div
    className="euiFlexItem euiFlexItem--flexGrowZero"
  >
    <div
      className="euiPopover euiPopover--anchorUpCenter euiPopover-isOpen"
    >
      <div
        className="euiPopover__anchor"
      >
        <button
          className="euiButtonIcon euiButtonIcon--primary euiButtonIcon--empty euiButtonIcon--xSmall"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonIcon__icon"
            focusable="false"
            height={16}
            role="img"
            style={null}
            viewBox="0 0 16 16"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          />
        </button>
      </div>
      <Portal
        containerInfo={
          <div>
            <div
              data-focus-guard="true"
              style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
              tabindex="-1"
            />
            <div
              data-focus-lock-disabled="disabled"
            >
              <div
                aria-describedby="random_html_id"
                aria-live="off"
                aria-modal="true"
                class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--noShadow euiPopover__panel euiPopover__panel--top euiPopover__panel-isOpen"
                data-autofocus="true"
                role="dialog"
                style="top: -16px; left: -22px; will-change: transform, opacity; z-index: 2000;"
                tabindex="0"
              >
                <div
                  class="euiPopover__panelArrow euiPopover__panelArrow--top"
                  style="left: 10px; top: 0px;"
                />
                <p
                  class="euiScreenReaderOnly"
                  id="random_html_id"
                >
                  You are in a dialog. To close this dialog, hit escape.
                </p>
                <div>
                  <h3
                    class="euiTitle euiTitle--xxsmall"
                  >
                    Pattern regex
                  </h3>
                  <div
                    class="euiText euiText--small"
                  >
                    Log patterns allow you to cluster your logs, to help
                  </div>
                  <div
                    class="euiText euiText--small"
                  >
                    summarize large volume of logs.
                  </div>
                  <div
                    class="euiSpacer euiSpacer--s"
                  />
                  <div
                    class="euiFormRow"
                    id="random_html_id-row"
                  >
                    <div
                      class="euiFormRow__fieldWrapper"
                    >
                      <div
                        class="euiFormControlLayout"
                      >
                        <div
                          class="euiFormControlLayout__childrenWrapper"
                        >
                          <input
                            aria-describedby="random_html_id-help-0"
                            class="euiFieldText"
                            id="random_html_id"
                            type="text"
                            value="[a-zA-Zd]"
                          />
                        </div>
                      </div>
                      <div
                        class="euiFormHelpText euiFormRow__text"
                        id="random_html_id-help-0"
                      >
                        <div
                          class="euiText euiText--small"
                        >
                          Pattern regex is used to reduce logs into log groups.
                           
                          <a
                            class="euiLink euiLink--primary"
                            href="https://github.com/opensearch-project/sql/blob/2.x/docs/user/ppl/cmd/patterns.rst#description"
                            rel="noopener noreferrer"
                            target="_blank"
                          >
                            help
                            <svg
                              aria-hidden="true"
                              aria-label="External link"
                              class="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
                              focusable="false"
                              height="16"
                              role="img"
                              viewBox="0 0 16 16"
                              width="16"
                              xmlns="http://www.w3.org/2000/svg"
                            />
                            <span
                              class="euiScreenReaderOnly"
                            >
                              (opens in a new tab or window)
                            </span>
                          </a>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="euiPopoverFooter"
                  >
                    <div
                      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
                    >
                      <div
                        class="euiFlexItem euiFlexItem--flexGrowZero"
                      >
                        <button
                          class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--small"
                          type="button"
                        >
                          <span
                            class="euiButtonContent euiButtonEmpty__content"
                          >
                            <span
                              class="euiButtonEmpty__text"
                            >
                              Cancel
                            </span>
                          </span>
                        </button>
                      </div>
                      <div
                        class="euiFlexItem euiFlexItem--flexGrowZero"
                      >
                        <button
                          class="euiButton euiButton--primary euiButton--small euiButton--fill"
                          type="button"
                        >
                          <span
                            class="euiButtonContent euiButton__content"
                          >
                            <span
                              class="euiButton__text"
                            >
                              Apply
                            </span>
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              data-focus-guard="true"
              style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
              tabindex="-1"
            />
          </div>
        }
      >
        Array [
          <div
            data-focus-guard={true}
            key="guard-first"
            style={
              Object {
                "height": "0px",
                "left": "1px",
                "overflow": "hidden",
                "padding": 0,
                "position": "fixed",
                "top": "1px",
                "width": "1px",
              }
            }
            tabIndex={-1}
          />,
          <div
            data-focus-lock-disabled="disabled"
            onBlur={[Function]}
            onFocus={[Function]}
            onScrollCapture={[Function]}
            onTouchMoveCapture={[Function]}
            onWheelCapture={[Function]}
          >
            <div
              aria-describedby="random_html_id"
              aria-live="off"
              aria-modal="true"
              className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--noShadow euiPopover__panel euiPopover__panel--top euiPopover__panel-isOpen"
              role="dialog"
              style={
                Object {
                  "left": -22,
                  "top": -16,
                  "willChange": "transform, opacity",
                  "zIndex": 2000,
                }
              }
              tabIndex={0}
            >
              <div
                className="euiPopover__panelArrow euiPopover__panelArrow--top"
                style={
                  Object {
                    "left": 10,
                    "top": 0,
                  }
                }
              />
              <p
                className="euiScreenReaderOnly"
                id="random_html_id"
              >
                You are in a dialog. To close this dialog, hit escape.
              </p>
              <div>
                <h3
                  className="euiTitle euiTitle--xxsmall"
                >
                  Pattern regex
                </h3>
                <div
                  className="euiText euiText--small"
                >
                  Log patterns allow you to cluster your logs, to help
                </div>
                <div
                  className="euiText euiText--small"
                >
                  summarize large volume of logs.
                </div>
                <div
                  className="euiSpacer euiSpacer--s"
                />
                <div
                  className="euiFormRow"
                  id="random_html_id-row"
                >
                  <div
                    className="euiFormRow__fieldWrapper"
                  >
                    <div
                      className="euiFormControlLayout"
                    >
                      <div
                        className="euiFormControlLayout__childrenWrapper"
                      >
                        <input
                          aria-describedby="random_html_id-help-0"
                          className="euiFieldText"
                          id="random_html_id"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          type="text"
                          value="[a-zA-Zd]"
                        />
                      </div>
                    </div>
                    <div
                      className="euiFormHelpText euiFormRow__text"
                      id="random_html_id-help-0"
                    >
                      <div
                        className="euiText euiText--small"
                      >
                        Pattern regex is used to reduce logs into log groups.
                         
                        <a
                          className="euiLink euiLink--primary"
                          href="https://github.com/opensearch-project/sql/blob/2.x/docs/user/ppl/cmd/patterns.rst#description"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          help
                          <svg
                            aria-hidden={true}
                            aria-label="External link"
                            className="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
                            focusable="false"
                            height={16}
                            role="img"
                            style={null}
                            viewBox="0 0 16 16"
                            width={16}
                            xmlns="http://www.w3.org/2000/svg"
                          />
                          <span
                            className="euiScreenReaderOnly"
                          >
                            (opens in a new tab or window)
                          </span>
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="euiPopoverFooter"
                >
                  <div
                    className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
                  >
                    <div
                      className="euiFlexItem euiFlexItem--flexGrowZero"
                    >
                      <button
                        className="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--small"
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                      >
                        <span
                          className="euiButtonContent euiButtonEmpty__content"
                        >
                          <span
                            className="euiButtonEmpty__text"
                          >
                            Cancel
                          </span>
                        </span>
                      </button>
                    </div>
                    <div
                      className="euiFlexItem euiFlexItem--flexGrowZero"
                    >
                      <button
                        className="euiButton euiButton--primary euiButton--small euiButton--fill"
                        disabled={false}
                        onClick={[Function]}
                        style={
                          Object {
                            "minWidth": undefined,
                          }
                        }
                        type="button"
                      >
                        <span
                          className="euiButtonContent euiButton__content"
                        >
                          <span
                            className="euiButton__text"
                          >
                            Apply
                          </span>
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>,
          <div
            data-focus-guard={true}
            style={
              Object {
                "height": "0px",
                "left": "1px",
                "overflow": "hidden",
                "padding": 0,
                "position": "fixed",
                "top": "1px",
                "width": "1px",
              }
            }
            tabIndex={-1}
          />,
        ]
      </Portal>
    </div>
  </div>
</div>
`;
